tests/binary-with-tool/BUILD.bazel

```diff
@@ -0,0 +1,17 @@
+load(
+    "@io_tweag_rules_haskell//haskell:haskell.bzl",
+    "haskell_test",
+)
+
+package(default_testonly = 1)
+
+haskell_test(
+    name = "binary-with-tool",
+    srcs = ["Main.hs"],
+    args = ["$(location //tests/binary-simple)"],
+    expected_covered_expressions_percentage = 100,
+    tags = ["coverage-compatible"],
+    tools = ["//tests/binary-simple"],
```

Comment:
This test case confuses me. I understand the `tools` attribute as compile time dependencies, plugins and preprocessors. But, then using it as a `$(location ...)` arg sounds more like a `data` attribute dependency. The test-case also doesn't actually use the tool at compile time, so I'm not sure what it really tests.

Reply:
It passes the tool as an argument to the binary. The binary checks that there is an argument. If there is none, it fails. Not that we would get that far if `tools` didn't exist: because then `$(location ...)` would fail.

Reply:
My point is it doesn't test anything about the handling of `tools` in rules_haskell. To illustrate this:
- if the attribute is changed to `data = ["//tests/binary-simple"]`, then the test-case still passes.
- also if `_resolve_preprocessors` is replaced by the following dummy, then the test-case still passes. 
    ```
    def _resolve_preprocessors(ctx, preprocessors):
        return struct(inputs = depset(), input_manifests = [])
    ```